add_me_to_your_PATH/Dockerfile.template

```diff
@@ -53,61 +83,64 @@ RUN --mount=type=secret,id=github_token \
               Pkg.Registry.add("General")'
 
+#####
+##### `sysimage-image` stage
+#####
 
-FROM base as sysimage-image
+# The sysimage stage is designed to be invalidated as infrequently as
+# possible while making a sysimage for faster Julia load times.
 
-# Install system dependencies needed to instantiate the environment and build
-# sysimage. Based on Docker's best practices w.r.t. apt-get:
-# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
+FROM base as sysimage-image
 
-RUN apt-get update && \
-    apt-get install -y gcc && \
+# Installing required system packages:
+# - `gcc`: Required for PackageCompiler.jl
+RUN apt-get -qq update && \
+    apt-get -qq install gcc && \
     rm -rf /var/lib/apt/lists/*
 
 # Instantiate the Julia project environment
-COPY --from=sysimage-project *Project.toml *Manifest.toml /JuliaProject/
+ENV JULIA_PROJECT /JuliaProject
+COPY --from=sysimage-project Project.toml Manifest.toml ${JULIA_PROJECT}/
 
 RUN --mount=type=secret,id=github_token \
-    julia --project=/JuliaProject -e 'using Pkg; Pkg.Registry.update(); Pkg.instantiate(); Pkg.build(); Pkg.precompile()'
+    julia -e 'using Pkg; Pkg.Registry.update(); Pkg.instantiate(); Pkg.build(); Pkg.precompile(strict=true)'
 
-COPY ./julia_pod/sysimage.jl /JuliaProject/sysimage.jl
+COPY ./julia_pod/sysimage.jl ${JULIA_PROJECT}/sysimage.jl
+ARG SYSIMAGE="true"
 RUN --mount=type=secret,id=github_token \
-    julia --project=/JuliaProject -e 'include("/JuliaProject/sysimage.jl")'
+    if [ "$SYSIMAGE" = "true" ]; then \
+        julia ${JULIA_PROJECT}/sysimage.jl; \
+    fi
 
 RUN --mount=type=secret,id=github_token \
-    julia --project=/JuliaProject -e 'using Pkg; Pkg.instantiate()'
+    julia -e 'using Pkg; Pkg.instantiate()'
 
 #####
-##### sysimage stage (above)
+##### `precompile-image` stage
 #####
-# The sysimage stage is designed to be invalidated as infrequently as
-# possible while making a sysimage for faster julia load times.
-# In particular, it only depends on the 'Manifest.toml' containing
-# pinned dependencies only, which by julia_pod convention
-# is the set of dependencies that will go into the sysimage.
-
 
 FROM sysimage-image as precompile-image
 
-# install Revise and PProf, even if not a dep of project
+# Optionally install the Revise and PProf packages without adding it to the project
 ARG ADD_UTILS="true"
 RUN --mount=type=secret,id=github_token \
     if [ "$ADD_UTILS" = "true" ]; then \
-        julia -e 'using Pkg; Pkg.add("Revise"; preserve=Pkg.PRESERVE_ALL); Pkg.add("PProf"; preserve=Pkg.PRESERVE_ALL); Pkg.instantiate()'; \
+        JULIA_PROJECT="" julia -e 'using Pkg; Pkg.add(["Revise", "PProf"]; preserve=Pkg.PRESERVE_ALL)'; \
     fi
 RUN mkdir -p /root/.julia/config
 
-COPY *Project.toml *Manifest.toml /JuliaProject/
+COPY *Project.toml *Manifest.toml ${JULIA_PROJECT}/
 
 # comment out if you don't have any `dev --local` deps
-COPY dev/ /JuliaProject/dev/
+COPY dev* ${JULIA_PROJECT}/dev/
```

Comment:
I can never remember how glob matching like this works...this will copy everything starting with `dev` into the directory `dev/` on teh container right?  even things like `devstuff/things`?

Reply:
That is correct. In the past the `COPY` command needed to copy at least one file but with the `dockerfile:1.4` that doesn't seem to be a requirement. This change means that this line no longer needs to be commented out if the `dev` directory isn't present. 

Alternatively, I could do `COPY de? ${JULIA_PROJECT}/dev/` to copy over any directories matching this regex `/^de.$/` but that seemed more confusing